### ISSUE TITLE
Calico infos in etcd deleted automatically, causes bird_cfg_template config failed, key does not exist in path

### ISSUE THREAD

**Issue by bretagne-peiqi** [projectcalico/felix#2079]
## Context

In our product environment,  we use Calico BGP mesh to mesh mode (containerized) , I found BGP peers are all reset and keep in Socket Active state.  and saw calico-node logs as follows. 

I tried to restart all calico-node pod. Then when I use calicoctl node status: they all keep in established for several minutes, I found calico-node is using bird trying to configure route info in each node. it configure some routes but suddenly clean all routes and in etcd nodes infos are cleaned also.

I restarted again and it remains in same situation. 

## Your Environment

Calico version 3.1.1 image.
Kubernetes 1.11.2 version
etcd version 3.2.16
Operator system Centos 7.2 kernel 4.17
Our cluster has 17 network nodes and using mesh to mesh mode

![Screen Shot 2019-07-11 at 11 18 04 AM](https://user-images.githubusercontent.com/5621794/61193878-a8734800-a6f0-11e9-921f-b9b89129345f.png)

**Comment by bretagne-peiqi** [comment]
As it cleans all our routes in each nodes: routes inter nodes (proto bird); and routes intra nodes (via calixxx). Our network totally crashed...

**Comment by rafaelvanoni** [comment]
Can you verify that etcd is still in a healthy status? `calicoctl get node -o yaml` would help as well

**Comment by bretagne-peiqi** [comment]
hello, thanks for your comment, I verified Etcd status and it is good. calicoctl get node -o yaml is also normal. As we have run calico for more than one year,  I am suspecting if there is some bad calico data in etcd and when calico controller reconfigure data in Etcd, It causes conflicts and calico-node detects and clears all calico data. But I am not familiar with calico design and source code. so I am hoping to get some support.

Currently, I am trying to reproduce the situation, but I think it is not easy.

**Comment by tmjd** [comment]
If `calicoctl get node -o yaml` is normal, then that should be the same data that is used in calico-node for configuring bird (with confd).  Can you include the output of `calicoctl get node -o yaml` please?

Stepping back on the problem a bit, is this a persistent problem or a temporary issue? If it was temporary, was this due to perhaps a removed k8s node that was being cleaned up?

Another thing to look at would be the calico-kube-controller logs, the current version will clean up nodes in the datastore that have been removed from kubernetes. You could see if you see any logs indicating it did that for your nodes.
I don't believe calico-node will clear any calico node data, I think it will only update the node configuration for the node it is operating on.

**Comment by bretagne-peiqi** [comment]
@tmjd It's a temporary issue. we have rebuild our cluster and we have not kept the trace. 

calicoctl get node -o yaml is normal when calico-node is trying to configure routes, then one moment, I  saw the nodes infos in etcd disappeared, using calico get nodes -o wide, all nodes disappeared... so calicoctl get node -o yaml can neither see anything.  We have not remove or add new nodes in our cluster, so I don't think it's an operational mistake.

anyway, currently we have solved it by rebuild our cluster.  as it's rare, I have kept using calico about 3 years and first time saw such problem, I am curious about why it suddenly clear all hosts info or who clears all hosts info...

![Screen Shot 2019-07-11 at 11 47 02 AM](https://user-images.githubusercontent.com/5621794/61504246-47f44b80-aa0d-11e9-9c94-f6e77071772e.png)
![Screen Shot 2019-07-11 at 12 14 53 PM](https://user-images.githubusercontent.com/5621794/61504248-49be0f00-aa0d-11e9-8218-a00e69077f7b.png)
![Screen Shot 2019-07-11 at 11 42 58 AM](https://user-images.githubusercontent.com/5621794/61504251-4b87d280-aa0d-11e9-80f4-cb4a48d1b66e.png)
